core/templates/dev/head/pages/exploration_editor/editor_tab/StateContentEditorDirective.js

```diff
@@ -59,6 +60,11 @@ oppia.directive('stateContentEditor', [
 
           var saveContent = function() {
             stateContentService.saveDisplayedValue();
+            if (!$scope.isQuestion()) {
```

Comment:
Here and elsewhere, we should avoid doing stuff like branching on $scope.isQuestion().

Instead, call something like $scope.onStateContentChanged() or $scope.postSaveContentHook() and have that function be passed in as an attr to the directive. Move the stuff in the "if" to that function. That way you can have different post-hook functions based on the context, and the content service doesn't need to know what context it is operating in. Ditto below.

Reply:
Ok, yeah, that sounds good. Will do that. In that case, the passing of ExplorationStatesService to the directive would also be not needed.

Reply:
Done.